### PR TITLE
Test platform version changes for DTA

### PR DESCRIPTION
**Pull request by kaadhina**

### CODE REVIEW

File: Tasks/VsTest/distributedTest.ts [hunk 1]
```diff
@@ -53,6 +54,11 @@ export class DistributedTest {
             this.dtaTestConfig.vsTestVersion = '15.0';
         }
         utils.Helper.addToProcessEnvVars(envVars, 'DTA.TestPlatformVersion', this.dtaTestConfig.vsTestVersion);
+        utilities.locateTestWindow(this.dtaTestConfig)
```
Comment (RanjanarMS) [inline Tasks/VsTest/distributedTest.ts]: await wont work here?

File: Tasks/VsTest/settingsHelper.ts [hunk 2]
```diff
@@ -65,14 +65,14 @@ export async function updateSettingsFileAsRequired(settingsFile: string, isParal
...
-            tl.warning(tl.loc('FailedToSetRunConfiguration'));
+            tl.warning(tl.loc('InvalidSettingsFile', settingsFile));
             settingsExt = null;
         }
     } else if (settingsFile && utilities.pathExistsAsFile(settingsFile)) {
         settingsExt = runSettingsExt;
         result = await utilities.getXmlContents(settingsFile);
-        if(!result || result.RunSettings === undefined) {
-            tl.warning(tl.loc('FailedToSetRunConfiguration'));
+        if(!result || result.RunSettings === undefined) {
```
Comment (RanjanarMS) [inline Tasks/VsTest/settingsHelper.ts]: minor remove the whitespace

File: Tasks/VsTest/utilities.ts [hunk 3]
```diff
@@ -15,6 +15,108 @@ var parser = new xml2js.Parser();
 var builder = new xml2js.Builder();
 var headlessBuilder = new xml2js.Builder({headless: true});
 
+
```
Comment (RanjanarMS) [inline Tasks/VsTest/utilities.ts]: move all these to versionFinder.ts
utils is becoming big

File: Tasks/VsTest/versionFinder.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            deferred.resolve(vstestConsoleInfo);
+        });
+    return deferred.promise;
+}
+
+export function locateTestWindow(testConfig: models.TestConfigurations): Q.Promise<models.ExecutabaleInfo> {
+    let deferred = Q.defer<models.ExecutabaleInfo>();
+    let vsVersion: number = parseFloat(testConfig.vsTestVersion);
+
+    if (isNaN(vsVersion) || vsVersion === 15.0) {
```
Comment (cltshivash) [inline Tasks/VsTest/versionFinder.ts]: > === 15.0 [](start = 37, length = 9)

can we avoid the hardcoding ?

File: Tasks/VsTest/versionFinder.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+import Q = require('q');
+import models = require('./models')
+
+var regedit = require('regedit');
+var xml2js = require('xml2js');
+
+export function locateVSTestConsole(testConfig): Q.Promise<models.ExecutabaleInfo> {
+    let deferred = Q.defer<models.ExecutabaleInfo>();
+    locateTestWindow(testConfig).
+        then( function(exeInfo) {
```
Comment (cltshivash) [inline Tasks/VsTest/versionFinder.ts]: >then [](start = 8, length = 4)

+1 on Ranjan's comment.. why are we doing a then here and not async model..

File: Tasks/VsTest/versionFinder.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
+import tl = require('vsts-task-lib/task');
+import path = require('path');
+import Q = require('q');
+import models = require('./models')
+
+var regedit = require('regedit');
+var xml2js = require('xml2js');
+
+export function locateVSTestConsole(testConfig): Q.Promise<models.ExecutabaleInfo> {
```
Comment (cltshivash) [inline Tasks/VsTest/versionFinder.ts]: >locateVSTestConsole [](start = 16, length = 19)

registry based locator should be a separate class and willow should be separate class..isn't it possible ?

**Pull request merged**